lib/SILOptimizer/Utils/Differentiation/PullbackEmitter.cpp

```diff
@@ -510,6 +512,232 @@ void PullbackEmitter::addToAdjointBuffer(SILBasicBlock *origBB,
   accumulateIndirect(adjointBuffer, rhsBufferAccess, loc);
 }
 
+//--------------------------------------------------------------------------//
+// Member accessor pullback generation
+//--------------------------------------------------------------------------//
+
+/// Returns true if the given original function is a "semantic member accessor".
+static bool isSemanticMemberAccessor(SILFunction *original) {
+  auto *dc = original->getDeclContext();
+  if (!dc)
+    return false;
+  auto *decl = dc->getAsDecl();
+  if (!decl)
+    return false;
+  auto *accessor = dyn_cast<AccessorDecl>(dc->getAsDecl());
+  if (!accessor)
+    return false;
+  // Currently, only getters and setters are supported.
+  // TODO(SR-12640): Support `modify` accessors.
+  if (accessor->getAccessorKind() != AccessorKind::Get &&
+      accessor->getAccessorKind() != AccessorKind::Set)
+    return false;
+  // Currently, only implicit accessors are supported.
```

Comment:
implicit accessors include getters/setters synthesized for stored properties. I'm not sure checking for implicitness is enough. I think we should check whether `AccessorDecl`'s parent decl is a `VarDecl` that has a property wrapper on it.

Reply:
I made the check more explicit in https://github.com/apple/swift/pull/31173/commits/49c4441fc48762261467e7d3893fb6942c051ab1.

I intentionally included stored property accessors to validate the new code path, and because the new code path has more room for micro-optimization. But I'm fine with reverting it.

Reply:
Another idea: we can lift the following check into `isSemanticMemberAccessor`: "`TangentVector` must be a struct that contains a stored property with the same name and corresponding `TangentVector` type of the given property".

If we do this, the new code path will be triggered only if it is guaranteed not to emit a non-differentiability error, which may slightly improve diagnostic locations. I'd prefer to investigate this in a follow-up.